tests/test_channel_properties.py

```diff
@@ -0,0 +1,197 @@
+"""Tests for validating channel properties for different basic data types."""
+import random
+
+import pytest
+
+import nidaqmx
+from nidaqmx.constants import (
+    BridgeConfiguration,
+    CounterFrequencyMethod,
+    Edge,
+    ExcitationSource,
+    FrequencyUnits,
+    PowerIdleOutputBehavior,
+    RTDType,
+    ResistanceConfiguration,
+    TemperatureUnits,
+    TerminalConfiguration,
+    TimeUnits,
+    VoltageUnits,
+)
+from nidaqmx.tests.helpers import generate_random_seed
+
+
+class TestChannelPropertyDataTypes(object):
+    """Contains a collection of pytest tests.
+
+    This validates the property getter,setter and deleter methods for different
+    data types of channel properties.
+    """
+
+    @pytest.mark.parametrize("seed", [generate_random_seed()])
+    def test_boolean_property(self, any_x_series_device, seed):
+        """Test for validating the channel property of boolean type."""
+        # Reset the pseudorandom number generator with seed.
+        random.seed(seed)
+
+        ai_phys_chan = random.choice(any_x_series_device.ai_physical_chans).name
+
+        with nidaqmx.Task() as task:
+            ai_channel = task.ai_channels.add_ai_voltage_chan_with_excit(
+                ai_phys_chan,
+                name_to_assign_to_channel="VoltageExcitChannel",
+                terminal_config=TerminalConfiguration.NRSE,
+                min_val=-10.0,
+                max_val=10.0,
+                units=VoltageUnits.VOLTS,
+                bridge_config=BridgeConfiguration.NO_BRIDGE,
+                voltage_excit_source=ExcitationSource.EXTERNAL,
+                voltage_excit_val=0.1,
+                use_excit_for_scaling=False,
+                custom_scale_name="",
+            )
+
+            # Test property initial value.
+            assert not ai_channel.ai_excit_use_for_scaling
+
+            # Test property setter and getter.
+            ai_channel.ai_excit_use_for_scaling = True
+            assert ai_channel.ai_excit_use_for_scaling
+
+            # Test property deleter.
+            del ai_channel.ai_excit_use_for_scaling
+            assert not ai_channel.ai_excit_use_for_scaling
+
+    @pytest.mark.parametrize("seed", [generate_random_seed()])
+    def test_enum_property(self, sim_ts_power_device, seed):
+        """Test for validating channel property of enum type."""
+        # Reset the pseudorandom number generator with seed.
+        random.seed(seed)
+
+        pwr_phys_chan = f"{sim_ts_power_device.name}/power"
+        voltage_setpoint = random.random() * 6.0
+        current_setpoint = random.random() * 3.0
+        output_enable = random.choice([True, False])
```

Comment:
these random values aren't valuable, just use a constant or even leverage the function default, if it exists

Reply:
Replaced random values with default.